tensorflow/compiler/xla/service/gpu/nvptx_compiler.cc

```diff
@@ -222,6 +223,37 @@ void WarnIfBadDriverJITVersion() {
   });
 }
 
+// Try to load ptx from files defined in the FLAGS. If successful, return true.
+bool MaybeLoadPtxFromFile(const HloModule* module, std::string* ptx) {
+  // If the xla_gpu_ptx_file options is set, be explicit when a file is used
+  // and warn when a file is not used to ease catching typo in filename.
+  std::string prefix = xla::FilenameFor(*module, *ptx);
+  std::string matched_filename;
+  for (const string filename : module->config().debug_options().xla_gpu_ptx_file()) {
+    // To ease comparing many PTX versions, accept different suffixes then
+    // the original filename.
+    if(absl::StartsWith(filename, prefix)) {
+      matched_filename = filename;
+      VLOG(0) << "RunBackend() - Will load PTX from file: " << filename;
+      break;
+    }
+  }
+  if (module->config().debug_options().xla_gpu_ptx_file().size() > 0 &&
+      matched_filename.empty()) {
+    VLOG(0) << "RunBackend() - For module with prefix '" << prefix
+            << "', we did not found a PTX file to load.";
```

Comment:
Regarding VLOG levels: VLOG's purpose is to help understand the program flow at different levels of detail. High log levels are used for high level activities, like "running buffer assignment" or alike. Hence, VLOG(0) doesn't seem right here.
What we want is to make the user aware of bad flags. That's important. Thinking about this again, the best way to do that is LOG(ERROR) or even LOG(FATAL). What do you think?

Reply:
I would need more complicated logic. Do we really want that for such a development feature?
I already error when we have an empty or non-existing file.
Currently, if a module doesn't have a file, does it mean no file was provided or one file with a bad name?

The other option would be to let the user specify explicitly for which module which file is.
But then we would also need to detect when a module was specified but not created.

All option for better error seem more involved and personally I do not think this is worthwhile for such not frequently feature. If you think otherwise, tell me.

Reply:
Fair enough.